### PR TITLE
Use mysql for ironic

### PR DESCRIPTION
**Pull request by shweta50**

### CODE REVIEW

File: scripts/runmariadb [hunk 1]
```diff
@@ -1,62 +0,0 @@
-#!/usr/bin/bash
```
Comment (poojaghumre) [inline scripts/runmariadb]: no need to delete this file from repo.. its unused anyway

File: scripts/createdb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+mysql -h mysql -u root -p${MYSQL_ROOT_PASSWORD} <<SQL_QUERY
+ CREATE USER IF NOT EXISTS 'ironic'@'localhost' IDENTIFIED BY '$MYSQL_PASSWORD' ;
+ CREATE USER IF NOT EXISTS 'ironic'@'%' IDENTIFIED BY '$MYSQL_PASSWORD' ;
```
Comment (poojaghumre) [inline scripts/createdb]: you just need to create one user "ironic".. no need for 2 users with @ part

File: scripts/createdb [hunk 3]
```diff
@@ -0,0 +1,9 @@
...
+ GRANT ALL on *.* TO 'ironic'@'localhost' IDENTIFIED BY '$MYSQL_PASSWORD' WITH GRANT OPTION ;
```
Comment (poojaghumre) [inline scripts/createdb]: can remove this line.. also fix remove extra whitespaces in this file at line beginnings

File: ironic-config/ironic.conf.j2 [hunk 4]
```diff
@@ -70,11 +70,7 @@ send_sensor_data_interval = 160
 bootloader = http://{{ env.IRONIC_URL_HOST }}:{{ env.HTTP_PORT }}/uefi_esp.img
 
 [database]
-{% if env.MARIADB_TLS_ENABLED == "true" %}
-connection = mysql+pymysql://ironic:{{ env.MARIADB_PASSWORD }}@127.0.0.1/ironic?charset=utf8&ssl=on&ssl_ca={{ env.MARIADB_CACERT_FILE }}
-{% else %}
-connection = mysql+pymysql://ironic:{{ env.MARIADB_PASSWORD }}@127.0.0.1/ironic?charset=utf8
-{% endif %}
+connection = mysql+pymysql://ironic:{{ env.MYSQL_PASSWORD }}@mysql/ironic?charset=utf8
```
Comment (poojaghumre) [inline ironic-config/ironic.conf.j2]: @mysql should be replaced with mysql host name or ip.. so @localhost or @127.0.0.1 would be appropriate here

File: scripts/createdb (hunk 3 above)
Comment (shweta50) [reply scripts/createdb]: Fixed

File: ironic-config/ironic.conf.j2 (hunk 4 above)
Comment (shweta50) [reply ironic-config/ironic.conf.j2]: Using the DNS name(mysql-svc) of MySQL service as the hostname.

File: scripts/runmariadb [hunk 5]
```diff
@@ -60,3 +60,4 @@ EOSQL
 else
     exec bash -x /usr/bin/mysqld_safe
 fi
+
```
Comment (poojaghumre) [inline scripts/runmariadb]: remove empty line

File: scripts/createdb [hunk 6]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+mysql -h mysql-svc -u root -p${MYSQL_PASSWORD} <<SQL_QUERY
+CREATE USER IF NOT EXISTS 'ironic' IDENTIFIED BY '$MYSQL_PASSWORD';
+GRANT ALL on *.* TO 'ironic' IDENTIFIED BY '$MYSQL_PASSWORD' WITH GRANT OPTION;
+CREATE DATABASE IF NOT EXISTS  ironic;
```
Comment (poojaghumre) [inline scripts/createdb]: remove extra ws before 'ironic'

**Review by poojaghumre — CHANGES_REQUESTED** [review]

**Review by poojaghumre — APPROVED** [review]
Thanks Shweta!

**Pull request merged**